benchmark/datasets.py

```diff
@@ -85,6 +85,21 @@ def xbin_mmap(fname, dtype, maxn=-1):
         n = min(n, maxn)
     return np.memmap(fname, dtype=dtype, mode="r", offset=8, shape=(n, d))
 
+#GW - I may eventually remove these
+def xbin_mmap_floatarray(fname, shape):
+    x = np.memmap(fname, dtype=numpy.float32, mode="r", offset=0, shape=shape)
+    return x
+
+def xbin_mmap_with_cast(fname, r_dtype, w_dtype, maxn=-1):
+    """ mmap the competition file format for a given type of items """
+    n, d = map(int, np.fromfile(fname, dtype="uint32", count=2))
+    assert os.stat(fname).st_size == 8 + n * d * np.dtype(r_dtype).itemsize
+    n = max(n, maxn)
+    x = np.memmap(fname, dtype=r_dtype, mode="r", offset=8, shape=(n, d))
+    x = x.astype( w_dtype )
+    return x
+#GW
```

Comment:
These seem to be used in the dataset creation. I'm not sure why T1/T2/T3 should use different abstractions on the datasets?

Reply:
These were useful to me at some point, but not needed anymore.  Removing them.